packages/react-storage/src/components/StorageBrowser/views/LocationActionView/UploadControls.tsx

```diff
@@ -214,15 +237,20 @@ export const UploadControls = (): JSX.Element => {
   const { direction, selection } = sortState;
 
   const tableData = tasks
-    .map(({ data, ...task }) => {
-      const { webkitRelativePath, type = '-' } = data;
+    .map(({ key, id, item, remove: _remove, ...task }) => {
+      const { size, webkitRelativePath, type = '-' } = item;
 
       const folder =
         webkitRelativePath?.length > 0
           ? webkitRelativePath.slice(0, webkitRelativePath.lastIndexOf('/') + 1)
           : '/';
 
-      return { ...task, data, type, folder };
+      const remove = () => {
+        dispatchStoreAction({ type: 'REMOVE_FILE_ITEM', id });
+        _remove();
+      };
+
+      return { ...task, folder, key, progress: 0, remove, size, type };
```

Comment:
`progress` support is currently missing and will be addressed in a targeted follow to `useProcessTasks`